ChuckNorrisFacts/Domain/Service/FactService.swift

```diff
@@ -6,14 +6,27 @@
 //  Copyright © 2018 Wanis. All rights reserved.
 //
 
+import Alamofire
 import Foundation
 import RxSwift
-import Alamofire
 
-struct FactService {
-    func get() {
-        Alamofire.request("https://api.chucknorris.io/jokes/random").responseData { response in
-            
-        }
+protocol FactServiceProtocol {
+    func getFacts(term: String) -> Observable<Facts>
+}
+
+struct FactService: FactServiceProtocol {
+    func getFacts(term: String) -> Observable<Facts> {
+        return Alamofire.request("https://api.chucknorris.io/jokes/search?query=\(term)").rx.responseData()
+            .retry(1)
+            .timeout(60, scheduler: MainScheduler.instance)
+            .map { data in
+                do {
+                    return try JSONDecoder().decode(Facts.self, from: data)
+                } catch {
+                    throw CNError.parsePayloadError
+                }
```

Comment:
Seria interessante colocar essa função do map como uma função separada...
Com ela separada vc consegue testar essa parse de forma unitária.
E se tiver numa classe separada ainda, vc pode mockar o resultado desse map na hora de testar o `FactService`, conseguindo ver como o `FactService` se comporta na hora que vc força o map a cair no `throw`, por exemplo.

Reply:
Fora que fica mais legível o código :)

Reply:
Boa!